build/prepend_edoc_autogenerated

```diff
@@ -1,27 +1,28 @@
-#! /bin/sh
+#!/usr/bin/env escript
+%% -*- erlang -*-
 
-f="$1"
-orig="${2:-$f}"
-
-if (head -n 1 "$f" | grep -q '@doc automatically generated scanner')
-then
-    # already prepended
-    exit 0
-fi
-
-cleanup () {
-    rm -f "$f".tmp
-}
-
-trap 'ec=$?; cleanup; exit $ec' EXIT QUIT INT TERM
-
-set -e
-cat > "$f".tmp <<EOF
-%% @doc automatically generated scanner
+-define(HEAD_LINE, "@doc automatically generated scanner").
+-define(ALL_AUTO_STR_FORMAT, "%%" ++ ?HEAD_LINE ++ "
 %% @private
 %% @end
--file("$orig", 0).
-EOF
-cat "$f" >> "$f".tmp
-touch -r "$f" "$f".tmp
-mv "$f.tmp"  "$f"
+-file(\"~s\", 0).
+").
+
+
+main([ErlFile, OrigFile]) ->
+    case file:open(ErlFile, [read, write]) of
+        {ok, F} ->
+            {ok, Line} = file:read_line(F),
+            case re:run(Line, ?HEAD_LINE) of
+                'nomatch' ->
+                    {ok, ErlStr} = file:read_file(ErlFile),
+                    file:position(F, 0),
+                    file:write(F, io_lib:format(?ALL_AUTO_STR_FORMAT, [OrigFile]) ++ ErlStr);
+                _ ->
+                    ok
+            end,
+            file:close(F);
+        _ ->
+            io:format(standard_error, "file ~s open failed!~n", [ErlFile]),
```

Comment:
Do not throw away the `Reason`! It is valuable info when things go wrong. Use `file:format_error` to and include both the error message and the error code (reason) in the output.